course7/homeworks/patrickkostjens/complex/main.cc

```diff
@@ -0,0 +1,44 @@
+#include "calculator.hh"
+
+#include <boost/filesystem.hpp>
+#include <string>
+#include <dlfcn.h>
+
+namespace fs = boost::filesystem;
+
+int main(int argc, char** argv)
+{
+	Calculator c;
+	int plugin_count = 0;
+	for (const auto& it : fs::directory_iterator(fs::path(".")))
+		if (it.path().extension() == ".so" || it.path().extension() == ".dylib")
+			plugin_count++;
+
+	plugin *plugins = new plugin[plugin_count];
```

Comment:
You already store the function pointers in the calculator. Do you need to store them separately here?

Reply:
Before, it broke when I didn't do it, but I now find another way to do it where I don't need to store them here. Hopefully that's better. It at least looks simpler.